research-software-directory/RSD-as-a-service
# Add support for third-party services

Fixes #1288 

Changes proposed in this pull request:

* adds support for third-party services via plugin slots in the user interface. The plugin slots are links that are provided by third-party services
* supports an arbitrary amount of plugins
* currently implemented slots are: `userMenu` and `softwareEditNav` (see `type PluginSlot`)
* adds a documentation
* adds syntax highlighting in docusaurus for json, nginx and typescript
* adds a `data` attribute to the user token which contains `eduPersonEntitlements` if the user logs in via HelmholtzID

How to test:

* follow steps to setup example plugin container in [research-software-directory/RSD-plugin-example](https://github.com/research-software-directory/RSD-plugin-example) (in progress)
* login
* verify that plugin slots are shown including svg icons in the user menu and in the software edit navigation sidebar

PR Checklist:
